monitoring/metric.go

```diff
@@ -0,0 +1,131 @@
+package monitoring
+
+import (
+	"strings"
+	"time"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/cloudfoundry-community/splunk-firehose-nozzle/eventwriter"
+	"github.com/cloudfoundry-community/splunk-firehose-nozzle/utils"
+)
+
+const (
+	SplunkMetric = "metric_name:"
+)
+
+type Metrics struct {
+	CallerFuncs   map[string]MonitorFunc
+	Counters      map[string][]utils.Counter
+	logger        lager.Logger
+	interval      time.Duration
+	ticker        *time.Ticker
+	writer        eventwriter.Writer
+	filtermetrics string
```

Comment:
nit: use camelcase convention

created a set for filterMetric. It will reduce search time from O(N) to O(1)